### PR TITLE
sql-database: detect precision and scale of supported column types

### PR DESCRIPTION
**Pull request by steinitzu**
# Tell us what you do here

- [x] improving, documenting, customizing existing source (please link an issue or describe below)

Detects precision and scale attributes for reflected columns in the source database.

Types supported:

`int/bigint/smallint` -> `bigint` with corresponding precision
`numeric` -> `decimal` with precision (sqlalchemy `Numeric` is base of all "decimal" like types)
`timestamp/time` -> dlt `timestamp/time` with precision
`string/binary` -> `text`/`binary` with length as precision

Detects types using the base type classes in `sqlalchemy.sql.sqltypes` so should work for all dialects.

# Relevant issue

resolves  https://github.com/dlt-hub/verified-sources/issues/323

### CODE REVIEW

**Comment by steinitzu** [comment]
The rfam db is down I think, why these tests are failing

```
FAILED tests/sql_database/test_sql_database_source.py::test_load_mysql_data_load[postgres] - sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (1045, "Access denied for user 'rfamro'@'172.183.107.247' (using password: NO)")
(Background on this error at: https://sqlalche.me/e/20/e3q8)
FAILED tests/sql_database/test_sql_database_source.py::test_load_mysql_data_load[duckdb] - sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (1045, "Access denied for user 'rfamro'@'172.183.107.247' (using password: NO)")
(Background on this error at: https://sqlalche.me/e/20/e3q8)
```

File: sources/sql_database/schema_types.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            name=sql_col.name,
+            data_type="decimal",
+            precision=sql_t.precision,
+            scale=sql_t.scale,
+        )
+    elif isinstance(sql_t, sqltypes.String):
+        col = dict(name=sql_col.name, data_type="text", precision=sql_t.length)
+    elif isinstance(sql_t, sqltypes._Binary):
+        col = dict(name=sql_col.name, data_type="binary", precision=sql_t.length)
+    elif isinstance(sql_t, sqltypes.DateTime):
```
Comment (rudolfix) [inline sources/sql_database/schema_types.py:35]: I'm not sure we should emit timestamp precision. it may just create incompatibilities and AFAIK sql alchemy uses datetime (floats) so precision is always microseconds anyway.
TLDR: I'd leave datetime and time precision

**Review by rudolfix — CHANGES_REQUESTED** [review]
looks good! pls check my comment about datetime
this mysql database sometimes is online sometimes not. so let's ignore it for now

File: sources/sql_database/schema_types.py (hunk 1 above)
Comment (steinitzu) [reply sources/sql_database/schema_types.py]: True, makes sense. I've removed time/datetime

**Review by rudolfix — APPROVED** [review]
LGTM!
